File: Task/Use-another-language-to-call-a-function/Nim/use-another-language-to-call-a-function-1.nim
proc Query*(data: var array[1024, char], length: var cint): cint {.exportc.} =
  const text = "Here am I"
  if length < text.len:
    return 0

  for i in 0 .. text.high:
    data[i] = text[i]
  length = text.len
  return 1

File: Task/Use-another-language-to-call-a-function/Lua/use-another-language-to-call-a-function-1.lua
-- Query.lua

--  basic kv dbase
function qdb_init ()
   qdb = {} -- global in lua
   qdb["who are you?"]   = "I am Lua"
   qdb["what are you?"]  = "I am a Lua virtual machine"
   qdb["when are you?"]  = "I am live"
   qdb["where are you?"] = "Here I am"
   qdb["how are you?"]   = "I am a fine"
   qdb["default"]        = "unknown query"

end

-- call from C with key => Data
function Query(k)

   local key  = string.lower(k)

   if (qdb[key] == nil) then key = "default" end

   lua_response = qdb[key]

end
